### ISSUE TITLE
AUR Build Error...

### ISSUE THREAD

**Issue by kinesivan** [rilian-la-te/vala-panel#26]
```
gobject-2.0.vapi:157.37-157.57: error: The type name `GLib.DuplicateFunc' could not be found
		public T dup_data<T> (string key, GLib.DuplicateFunc<T> dup_func);
		                                  ^^^^^^^^^^^^^^^^^^^^^
gobject-2.0.vapi:160.44-160.64: error: The type name `GLib.DuplicateFunc' could not be found
		public T dup_qdata<T> (GLib.Quark quark, GLib.DuplicateFunc<T> dup_func);
		                                         ^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 2 error(s), 49 warning(s)
lib/CMakeFiles/valapanel.dir/build.make:119: recipe for target 'lib/valapanel_valac.stamp' failed
make[2]: *** [lib/valapanel_valac.stamp] Error 1
CMakeFiles/Makefile2:270: recipe for target 'lib/CMakeFiles/valapanel.dir/all' failed
make[1]: *** [lib/CMakeFiles/valapanel.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Makepkg was unable to build vala-panel-git.
==> Restart building vala-panel-git ? [y/N]
==> ---------------------------------------
==> 

```
I'm using yaourt to build the package, but cloning and building from this repo yields the same result either way.

**Comment by rilian-la-te** [comment]
Remove gobject-2.0.vapi from repo

**Comment by Tomiaz** [comment]
I can't find that file in the repo, were you mentioning something else?

**Comment by rilian-la-te** [comment]
remove all glib -related vapis from vapi folder and try to build. If builds - I will add this to PKGBUILD

**Comment by Tomiaz** [comment]
Build successful, but I get an fatal error upon launching it:

`
(vala-panel:26407): GLib-GIO-CRITICAL **: g_menu_remove: assertion '0 <= position && position < menu->items->len' failed

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:27:30: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:33:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:39:30: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:40:27: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:43:28: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:47:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:27:30: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:33:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:39:30: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:40:27: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:43:28: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:47:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:27:30: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:33:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:39:30: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:40:27: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:43:28: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:47:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

** (vala-panel:26407): WARNING **: tray: another systray already running

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:27:30: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:33:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:39:30: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:40:27: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:43:28: The :prelight pseudo-class is deprecated. Use :hover instead.

(vala-panel:26407): Gtk-WARNING **: Theme parsing error: style.css:47:18: The '-gtk-image-effect' property has been renamed to '-gtk-icon-effect'

(vala-panel:26407): Gtk-ERROR **: GtkBox child Pager minimum width: -1 < 0 for height 1
[1]    26407 trace trap (core dumped)  vala-panel
`

**Comment by rilian-la-te** [comment]
Try using empty panel and adding plugins on demand. When crashes, write plugin name here

**Comment by snckrz** [comment]
I have the same build error, but I can't seem to get the files deleted. I tried doing a fork and removing  it there, but after some directory permission problems, I got an error that RPMtools, especially rpmbuild cant be found. Some help would really be appreciated! :)

**Comment by Tomiaz** [comment]
Encountered the same problem, but a simple edit solved my issue. In your PKGBUILD file in the archlinux folder, go to the prepare() function. Add `rm $(srcfolder)/vala-panel/vapi/glib-2.0.vapi`

$(srcfolder) may not be the right term, I apologise I can't test it right now.

**Comment by snckrz** [comment]
Nope. doesn't work. It deletes the files, but Im still getting following error:
`[snckrz@Andromeda archlinux]$ makepkg -sri
==> Making package: vala-panel-git 0.3.5.r9.g42ba3ba-1 (Tue May 17 23:14:39 CEST 2016)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating vala-panel git repo...
Fetching origin
  -> Updating cmake-vala git repo...
Fetching origin
  -> Updating vala-dbusmenu git repo...
Fetching origin
==> Validating source files with sha256sums...
    vala-panel ... Skipped
    cmake-vala ... Skipped
    vala-dbusmenu ... Skipped
==> Extracting sources...
  -> Creating working copy of vala-panel git repo...
Reset branch 'makepkg'
  -> Creating working copy of cmake-vala git repo...
Reset branch 'makepkg'
  -> Creating working copy of vala-dbusmenu git repo...
Reset branch 'makepkg'
==> Starting prepare()...
==> Starting pkgver()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
-- Looking for RPMTools... - rpmbuild NOT FOUND
-- checking for a minimum Vala version of 0.24
--   found Vala, version 0.32.0
-- Checking for modules 'gtk+-3.0>=3.12;glib-2.0>=2.40;gio-2.0>=2.40;gio-unix-2.0>=2.40;gthread-2.0>=2.40;libpeas-1.0>=1.2;libpeas-gtk-1.0>=1.2;gmodule-2.0>=2.40'
--   
--   
CMake Error at /usr/share/cmake-3.5/Modules/FindPkgConfig.cmake:363 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/FindPkgConfig.cmake:528 (_pkg_check_modules_internal)
  CMakeLists.txt:95 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/home/snckrz/Downloads/vala-panel-master/archlinux/src/vala-panel/CMakeFiles/CMakeOutput.log".
==> ERROR: A failure occurred in build().
    Aborting...
[snckrz@Andromeda archlinux]$ 
`

**Comment by andreyorst** [comment]
Well, I downloaded package from here and added "rm $srcdir/$_pkgname/vapi/glib-2.0.vapi" to the prepare section in PKBUILD. Worken dor me. Same thing with vala-panel-appmenu.

**Comment by andresthor** [comment]
I was having some of the troubles posted above. I had tried installing rpmtools but was still getting dependency errors.

The solution for me was installing `gobject-introspection-1.0 `and adding `rm $srcdir/$_pkgname/vapi/glib-2.0.vapi `to my PKGBUILD.

The bar then compiled and I was able to install it.
When starting the bar I got the error posted above, by Tomiaz.

    (vala-panel:26407): Gtk-ERROR **: GtkBox child Pager minimum width: -1 < 0 for height 1 [1] 26407 trace trap (core dumped) vala-panel

I removed the following plugin from the config and got the bar working:

    [6]
    plugin-type='pager'
    has-schema=false
    is-expanded=false
    position=6
